api/src/main/java/org/openmrs/module/attachments/obs/Attachment.java

```diff
@@ -22,8 +24,14 @@
 	
 	protected String comment = "";
 	
+	protected String bytesMimeType = null;
+	
+	protected ContentFamily bytesContentFamily = null;
+	
 	protected ComplexData complexData = null;
 	
+	ComplexDataHelper1_10 complexDataHelper1 = new ComplexDataHelper1_10();
```

Comment:
Remove this field, hence the constructor will look like:

```java
public Attachment(Obs obs, ComplexDataHelper complexDataHelper) {
	this(obs);

	setBytesMimeType(complexDataHelper.getContentType(obs.getComplexData()));
        setBytesContentFamily(AttachmentsContext.getContentFamily(complexDataHelper.getContentType(obs.getComplexData())));
 }
``` 

Or even better like:

```java
public Attachment(Obs obs, ComplexDataHelper complexDataHelper) {
	this(obs);

	bytesMimeType  = complexDataHelper.getContentType(obs.getComplexData());
	bytesContentFamily = AttachmentsContext.getContentFamily(complexDataHelper.getContentType(obs.getComplexData()));
}
```

Reply:
@samuelmale i had relied on the constructor to be used by spring to do the dependency injection but it was no,throwing a null pointer exception not understanding the `ComplexDataHelper` interface instance..
So i decided to manually use the  `ComplexDataHelper1_10 `instance that implements `ComplexDataHelper`